Include/cpython/object.h

```diff
@@ -263,6 +263,10 @@ struct _typeobject {
 
     destructor tp_finalize;
     vectorcallfunc tp_vectorcall;
+
+    /* INTERNAL USE ONLY! MODIFYING THIS CAN CRASH PYTHON! */
+    const Py_ssize_t tp_obj_offset; /* Offset from "PyObject *" pointer */
+    const Py_ssize_t tp_obj_size; /* Total memory allocation size */
```

Comment:
I don't see the point of adding a new member. You can just check for Py_TPFLAGS_USES_OPAQUE_OBJECT flag in functions like PyType_GenericAlloc(), _PyObject_SIZE() and _PyObject_VAR_SIZE(). If the flag is set, add sizeof(PyObject).

Reply:
See previous comment.